egs/gigaspeech/ASR/local/compute_fbank_gigaspeech.py

```diff
@@ -134,78 +140,113 @@ def compute_fbank_gigaspeech(args):
                     cut_set
-                except NameError:
-                    logging.info(f"Reading {partition} raw cuts from disk.")
-                    cut_set = CutSet.from_file(raw_cuts_path)
-                # Note this step makes the recipe different than LibriSpeech:
-                # Since recordings are long, the initial CutSet has very long
-                # cuts with a plenty of supervisions. We cut these into smaller
-                # chunks centered around each supervision, possibly adding
-                # acoustic context.
-                logging.info(
-                    f"About to split {partition} raw cuts into smaller chunks."
+                    + cut_set.perturb_speed(0.9)
+                    + cut_set.perturb_speed(1.1)
                 )
-                cut_set = cut_set.trim_to_supervisions(
-                    keep_overlapping=False,
-                    min_duration=None
-                    if args.context_window <= 0.0
-                    else args.context_window,
-                    context_direction=args.context_direction,
+            cut_set.to_file(raw_cuts_path)
+
+        cuts_path = output_dir / f"cuts_{partition}{ctx_suffix}.jsonl.gz"
+        if cuts_path.is_file():
+            logging.info(
+                f"{partition} already exists - skipping cutting into "
+                f"sub-segments."
+            )
+        else:
+            try:
+                # If we skipped initializing `cut_set` because it exists
+                # on disk, we'll load it. This helps us avoid re-computing
+                # the features for different variants of context windows.
+                cut_set
+            except NameError:
+                logging.info(f"Reading {partition} raw cuts from disk.")
+                cut_set = CutSet.from_file(raw_cuts_path)
+            # Note this step makes the recipe different than LibriSpeech:
+            # Since recordings are long, the initial CutSet has very long
+            # cuts with a plenty of supervisions. We cut these into smaller
+            # chunks centered around each supervision, possibly adding
+            # acoustic context.
+            logging.info(
+                f"About to split {partition} raw cuts into smaller chunks."
+            )
+            cut_set = cut_set.trim_to_supervisions(
+                keep_overlapping=False,
+                min_duration=None
+                if args.context_window <= 0.0
+                else args.context_window,
+                context_direction=args.context_direction,
+            )
+
+            if args.precomputed_features:
+                # Extract the features after cutting large recordings into
+                # smaller cuts.
+                # Note:
+                # we support very efficient "chunked" feature reads with
+                # the argument `storage_type=ChunkedLilcomHdf5Writer`,
+                # but we don't support efficient data augmentation and
```

Comment:
>                 # but we don't support efficient data augmentation and
>                 # feature computation for long recordings yet.

Is it still the case with `kaldifeat`?

Reply:
No, `kaldifeat` solves the problem of feature computation, but data augmentation is still an issue. This may actually be another reason why @wgb14 observed OOM (I remember running into this before).